app/models/leaderboard.rb

```diff
@@ -0,0 +1,29 @@
+class Leaderboard
```

Comment:
I welcome suggestions, it definitely feels like some kind of an object trying to emerge, and this iteration is a step above all the logic being in the controller and/or the view. ;)

Reply:
I think I'd probably put this in `./lib/presenters` and not where it is. Having it as a model makes me think I'd be able to interact with it like a model and that it would have a database table associated with it. This is about aggregating data from a model, for the purpose of being able to render it to the user.